comment/models/flags.py

```diff
@@ -0,0 +1,159 @@
+from collections import namedtuple
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from comment.models import Comment
+
+User = get_user_model()
+
+class Flag(models.Model):
+
+    STATES = getattr(settings, 'COMMENT_FLAG_STATES', [
+        (1, _('flagged')),
+        (2, _('flag rejected by moderator')),
+        (3, _('creator notified')),
+        (4, _('content removed by creator')),
+        (5, _('content removed by owner'))
+    ])
+
+    # Make a named tuple
+    State = namedtuple('State', ['value', 'state'])
+
+    # Construct the list of named tuples(can't use list comprehension, scope issues)
+    states = []
+    for state in STATES:
+        states.append(State(*state))
+
+    comment = models.ForeignKey(Comment, on_delete=models.CASCADE, related_name='flags')
+    count = models.PositiveIntegerField(default=0)
+    creator = models.ForeignKey(User, on_delete=models.CASCADE, related_name='flag_owners')
+    state = models.SmallIntegerField(choices=STATES, default=states[0].value)
+    moderator = models.ForeignKey(User, null=True, on_delete=models.SET_NULL, related_name='flag_moderators')
+
+    def increase_count(self):
+        """Increase flag count and save the model """
+        self.refresh_from_db()
+        self.count = models.F('count') + 1
+        self.save()
+
+    def decrease_count(self):
+        """Decrease flag count and save the model """
+        self.refresh_from_db()
+        if self.count > 0:
+            self.count = models.F('flag') - 1
+            self.save()
+
```

Comment:
```suggestion
@receiver(post_save, sender=Comment)
def create_flag_object(sender, instance, created, raw, using, update_fields, **kwargs):
    if created:
        Flag.objects.create(comment=instance)
```

Reply:
I have already added this logic in the comment model. I just forgot to change the name of the function that receives the signal.

Reply:
Sorry I didn't notice it. I prefer to move this function to the Flad models.

Reply:
We had added this signal in the comments for the reaction model. Hence, I followed the same procedure here.

Reply:
I already moved it to models.reactions.

Reply:
I got what you mean. I think the branch you are working on is not up to date 😮
To avoid circular import I moved them. You can make new file signals for such functions in it's worth.

Reply:
My commits were made before you pushed in the respective changes in the other branch. No issues, I will merge and do the needful.